### PR TITLE
Feature 4 migrate clkutil

### LINKED ISSUE #4: Migrate clkutil from clkhash
We need to migrate clkutil - command line tools for creating a job, uploading CLKs and submitting a linking job here.

### PR DESCRIPTION
**Pull request by JoyceXinyueWang**
In this pull request, we migrated the main CLI tools from `clkhash`. Specifically, we migrated the following two files:
* cli.py => clk_cli.py
* rest_client.py => rest_client.py

Also we migrated related tests and test data
* test_cli.py

Please merge #1 before reviewing this pull request. 

Close #4

### CODE REVIEW

File: client/clk_cli.py [hunk 1]
```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/env python3.6
```
Comment (wilko77) [inline client/clk_cli.py]: remove

File: client/clk_cli.py [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+from __future__ import print_function
```
Comment (wilko77) [inline client/clk_cli.py]: we don't need that if we do not support python 2

File: client/clk_cli.py [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+import os
+import shutil
+from multiprocessing import freeze_support
+
+import click
+import clkhash
+import client
+from clkhash import benchmark as bench, clk, randomnames, validate_data, describe as descr
+from clkhash.schema import SchemaError, validate_schema_dict, convert_to_latest_version
+from clkhash.backports import raise_from
```
Comment (wilko77) [inline client/clk_cli.py]: same here, this was for python2 support

File: client/rest_client.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
+import time
+
+import requests
+import clkhash
+from clkhash.backports import TimeoutError
```
Comment (wilko77) [inline client/rest_client.py]: we don't need that any more as well

File: requirements.txt [hunk 5]
```diff
@@ -1,7 +1,8 @@
...
-clkhash == 0.14.0
+clkhash == 0.15.0
 jsonschema == 3.0.2
 numpy == 1.17.3
 pandas == 0.25.2
 pytest == 5.2.1
-requests == 2.22.0
\ No newline at end of file
+requests == 2.22.0
+retrying>=1.3
```
Comment (wilko77) [inline requirements.txt]: why are all dependencies pinned to an exact version but retrying isn't?

File: tests/__init__.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+BAD_SCHEMA_V2_PATH = os.path.join(TESTDATA, 'bad-schema-v2.json')
+BAD_SCHEMA_V3_PATH = os.path.join(TESTDATA, 'bad-schema-v3.json')
+
+RANDOMNAMES_SCHEMA_PATH = os.path.join(
+    TESTDATA,
+    'randomnames-schema-v2.json'
+)
+
+SAMPLE_DATA_PATH_1 = os.path.join(TESTDATA, 'dirty_1000_50_1.csv')
+SAMPLE_DATA_PATH_2 = os.path.join(TESTDATA, 'dirty_1000_50_2.csv')
```
Comment (wilko77) [inline tests/__init__.py]: both the SAMPLE_DATA_PATH_i are not used anywhere.

File: tests/__init__.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+import clkhash
+
+TESTDATA = os.path.join(
+    os.path.dirname(__file__),
+    'testdata'
+)
+
+SIMPLE_SCHEMA_PATH = os.path.join(TESTDATA, 'simple-schema.json')
+
+SAMPLE_DATA_SCHEMA_PATH = os.path.join(TESTDATA, 'dirty-data-schema.json')
```
Comment (wilko77) [inline tests/__init__.py]: not used anywhere

File: tests/test_clkcli.py [hunk 8]
```diff
@@ -0,0 +1,873 @@
...
+        :return: The parsed JSON in the created output file.
+        :raises: AssertionError if the command's exit code isn't 0
+        :raises: json.decoder.JSONDecodeError if the output isn't json
+        """
+        logging.info(command)
+        output_str = self.run_command_capture_output(command)
+        return json.loads(output_str)
+
+
+@unittest.skipUnless("INCLUDE_CLI" in os.environ,
```
Comment (wilko77) [inline tests/test_clkcli.py]: all of those should go. We want to always test CLI.

File: client/rest_client.py (hunk 4 above)
Comment (JoyceXinyueWang) [reply client/rest_client.py]: Why don't we need TimeoutError?

File: tests/__init__.py (hunk 6 above)
Comment (JoyceXinyueWang) [reply tests/__init__.py]: We need them when testing good schema

File: tests/__init__.py (hunk 7 above)
Comment (JoyceXinyueWang) [reply tests/__init__.py]: This is used when testing good schema

File: client/rest_client.py (hunk 4 above)
Comment (wilko77) [reply client/rest_client.py]: we don't need the one from backports. python3 comes with TimeoutError as standard

File: tests/__init__.py (hunk 7 above)
Comment (wilko77) [reply tests/__init__.py]: SAMPLE_DATA_SCHEMA_PATH is not used anywhere

**Comment by JoyceXinyueWang** [comment]
Randomnames-schema.json is in `anonlink-client/client/data/randomnames-schema.json`. All tests passed in my laptop

**Comment by wilko77** [comment]
`anonlink-client/client/data/randomnames-schema.json` is not part of this branch. Maybe you forgot to push that?

File: tests/test_clkcli.py [hunk 9]
```diff
@@ -178,8 +178,6 @@ def test_bad_v3_schema(self):
         assert 'schema is not valid.' in result.output
 
 
-@unittest.skipUnless("INCLUDE_CLI" in os.environ,
```
Comment (wilko77) [inline tests/test_clkcli.py]: please remove *all* occurrences of this line in the file.

File: requirements.txt [hunk 10]
```diff
@@ -5,4 +5,4 @@ numpy == 1.17.3
 pandas == 0.25.2
 pytest == 5.2.1
 requests == 2.22.0
-retrying>=1.3
+retrying==1.3
```
Comment (wilko77) [inline requirements.txt]: retrying 1.3.0 doesn't install. please set to current 1.3.3

**Pull request merged**